### PR TITLE
Add support for custom wrappers

### PR DESCRIPTION
**Pull request by bryevdv**
This PR adds two new command line options `--wrapper` and `--wrapper-inner` to allow users to insert their own custom wrapper executables int the legate invocation chain. 

Both wrappers come after `bind.sh` so that `%%LEGATE_GLOBAL_RANK%%` can be expanded in the invocation.

### CODE REVIEW

File: legate/driver/args.py [hunk 1]
```diff
@@ -356,20 +356,48 @@
...
 
+other.add_argument(
+    "--wrapper",
+    dest="wrapper",
+    required=False,
+    action="append",
+    default=[],
+    help="Specify another executable (and any command-line arguments for that "
+    "executable) that will be responsible to indirectly invoke legate, "
+    "including any other wrappers. May contain the special string "
```
Comment (manopapad) [inline legate/driver/args.py:374]:
```suggestion
    "executable) to wrap the Legate executable invocation. This wrapper will "
    "come right after the launcher invocation, and will be passed the rest of "
    "the Legate invocation (including any other wrappers) to execute. May "
    "contain the special string "
```

suggested rewording, similar for `--wrapper-inner`

**Review by manopapad — APPROVED** [review]
Looks good. Optionally suggest rewording the help text a bit to make it clearer what this is expected to do.

File: legate/driver/args.py (hunk 1 above)
Comment (bryevdv) [reply legate/driver/args.py]: https://github.com/nv-legate/legate.core/pull/813/commits/dc345d6ed0a428c5e786b84efa836b52a87f92c7

**Review by manopapad — APPROVED** [review]
